src/main/kotlin/com/github/attacktive/troubleshootereditor/domain/item/ItemObject.kt

```diff
@@ -147,19 +146,39 @@ object ItemObject {
 		}
 	}
 
-	fun overwriteProperties(connection: Connection, itemsPerPosition: Map<EquipmentPosition?, List<Item>>) {
-		itemsPerPosition
-			.map {
-				val position = it.key!!
-				val items = it.value
+	fun overwriteProperties(url: String, itemsPerPosition: Map<EquipmentPosition?, List<Item>>) {
```

Comment:
> Function overwriteProperties is nested too deeply.

Codacy complains the function is too deeply nested, so I tried extracting the `items.forEach` part into a function:
```kotlin
fun overwriteProperties(url: String, itemsPerPosition: Map<EquipmentPosition?, List<Item>>) {
	Database.connect(url)

	transaction {
		addLogger(StdOutSqlLogger)
		itemsPerPosition.forEach {
			val position = it.key
			val items = it.value

			if (position == null) {
				logger.warn("An item whose position is ${null} is found.")
			} else {
				val options = position.options
				if (options != null) {
					if (options.size > 5) {
						logger.warn("You can have up to 5 options! Other than the first five are going to be silently ignored.")
					}

					applyOptions(items, options)
				}
			}
		}
	}
}

private fun applyOptions(items: List<Item>, options: List<Pair<ItemOption, Int>>) = items.forEach { item ->
	ItemProperties.applyDefaultChanges(item.id)

	options.mapIndexed { index, pair ->
		val nthOptions = PropertyMaster.getNthOptions(index + 1)

		ItemProperties.insert(item.id, nthOptions.first, pair.first.value)
		ItemProperties.insert(item.id, nthOptions.second, pair.second)
	}
}
```

It doesn't really improve code readability. :man_shrugging: